### ISSUE TITLE
"Complete" is false even when the ring is complete

### ISSUE THREAD

**Issue by sathyakumarseshachalam** [linkedin/Burrow#72]
So this is again me testing burrow to check if it works per my own understanding. If the interpretation is wrong please correct me.

- Created a topic `burrow_test_2` with 2 partitions and produced 10 messages per partition using sarama's ' `kafka-console-producer` 
-  Modified `/etc/burrow/config/burrow.cfg` (where I keep the configuration file) and made 
```
[lagcheck]
intervals=2
```
- Restarted burrow.
- Now I ran a single `KafkaConsoleConsumer` and read the topic `--from-beginning`. Consumer config has entries 
```
group.id=MyBurrowConsumer2
offsets.storage=kafka
auto.commit.offset=true
```

Now since I have consumed all messages and committed the offset as well, I assumed the ring would be complete and I would get `complete` as `true` - But when I see the end point `v2/kafka/alpha/consumer/MyBurrowConsumer2/status`, I could only see `complete` as `false`

What am I missing ?

**Comment by toddpalino** [comment]
Given that you have set `intervals` to 2, Burrow will require two offset commits for all partitions before the ring (which has 2 entries) is considered complete. It's entirely possible that the console consumer has committed only once here because of the low number of messages. What you can try is while the consumer is running, produce another batch of messages. This should trigger a second offset commit (definitely - I'd have to check the logic of the console consumer and the client to see if it's shortcutting offset commits when there is no change in the offset - it's not supposed to).

**Comment by sathyakumarseshachalam** [comment]
@toddpalino : Thanks, I verified by writing my own java client instead that did a manual offset commit and found that the ring was complete, but I quickly ran into another issue.  

When is lag per partition lag. In my case, I simulated a "STOP"ed consumer. So I keep producing messages, so my head Offset is at the moment 200 per partition and consumers have committed offsets upto 100 only. As expected the consumer group is in an error condition and each partition is in a a "STOP"ed condition. However `/v2/kafka/<cluster>/consumer/<group>/lag` shows the lag per partition to be 0. I was expecting this to be 100 per partition and 200 total lag. They happen to be 0 at the moment

**Comment by sathyakumarseshachalam** [comment]
@toddpalino : I think the cause of the issue is that Kafka is taking off consumer groups that haven't committed in a while, Am unable to find the exact setting that will help me override that setting. But thanks for responding. Great work on Burrow. Like it and we started using in production. Closing this issue
